mlflow/tracing/provider.py

```diff
@@ -87,34 +91,66 @@ def _setup_tracer_provider():
 
     tracer_provider = TracerProvider()
     tracer_provider.add_span_processor(processor)
+    _force_set_otel_tracer_provider(tracer_provider)
+
+
+def _force_set_otel_tracer_provider(tracer_provider):
+    """
+    Resetting internal flag used in OpenTelemetry. If we don't reset the flag,
+    set_tracer_provider() will be a no-op after the first call
+    in the application lifecycle.
```

Comment:
Can we add a comment to the opentelemetry's code ref for _TRACER_PROVIDER_SET_ONCE just for clarification?